src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -31,13 +31,21 @@ public class ParserUtil {
     public static final String MESSAGE_INVALID_INDEX = "Index is not a non-zero unsigned integer.";
     public static final String MESSAGE_INVALID_INDEXES =
             "There should be exactly two non-zero unsigned integer indexes provided.";
+
+    private static final String MAX_SIZE = "2147483647";
+
+    private static final String MESSAGE_OUT_OF_RANGE = "Index provided is greater than maximum number of records.";
     /**
      * Parses {@code oneBasedIndex} into an {@code Index} and returns it. Leading and trailing whitespaces will be
      * trimmed.
      * @throws ParseException if the specified index is invalid (not non-zero unsigned integer).
      */
     public static Index parseIndex(String oneBasedIndex) throws ParseException {
         String trimmedIndex = oneBasedIndex.trim();
+        System.out.println(trimmedIndex.compareTo(MAX_SIZE));
+        if (trimmedIndex.compareTo(MAX_SIZE) >= 0) {
```

Comment:
Does the string comparison with the trimmed index work as intended?

Reply:
Updated to include length comparison in addition to lexicographic to fix erroneous exceptions thrown